addons/web_editor/static/src/js/widgets/widgets.js

```diff
@@ -1520,6 +1521,39 @@ var LinkDialog = Dialog.extend({
             class: data.classes.replace(/float-\w+/, '') + ' o_btn_preview',
         }).html((data.label && data.label.length) ? data.label : data.url);
     },
+    /**
+     * @private
+     */
+    _adaptPageAnchor: function () {
+        var self = this;
+        var urlInputValue = this.$('#o_link_dialog_url_input').val();
+        var $pageAnchor = this.$('.o_link_dialog_page_anchor');
+        var isFromWebsite = _.str.startsWith(urlInputValue, '/');
+        var isEmpty = urlInputValue.length === 0;
```

Comment:
No need of that variable and the condition should be written `!urlInputValue.length` (convention)